execution_engine/src/core/engine_state/mod.rs

```diff
@@ -1635,17 +1637,26 @@ where
             .map_err(Error::from)
     }
 
-    pub fn put_trie(
+    pub fn put_trie_and_find_missing_descendant_trie_keys(
         &self,
         correlation_id: CorrelationId,
         trie: &Trie<Key, StoredValue>,
-    ) -> Result<(), Error>
+    ) -> Result<InsertedTrieKeyAndMissingDescendants, Error>
     where
         Error: From<S::Error>,
     {
-        self.state
+        let inserted_trie_key = self
+            .state
             .put_trie(correlation_id, trie)
-            .map_err(Error::from)
+            .map_err(Error::from)?;
```

Comment:
Is `map_err(Error::from)` necessary? Doesn't `?` do that automatically? (Also below.)

Reply:
Fixed in https://github.com/CasperLabs/casper-node/pull/816/commits/262770cabe6535ee8aa7fc93217a86d562039a4e